manager/src/main/java/com/redhat/service/smartevents/manager/api/user/validators/processors/BaseGatewayConstraintValidator.java

```diff
@@ -0,0 +1,85 @@
+package com.redhat.service.smartevents.manager.api.user.validators.processors;
+
+import java.lang.annotation.Annotation;
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
+import org.hibernate.validator.internal.engine.messageinterpolation.util.InterpolationHelper;
+
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.GatewayProviderException;
+import com.redhat.service.smartevents.infra.models.gateways.Gateway;
+import com.redhat.service.smartevents.infra.validations.ValidationResult;
+import com.redhat.service.smartevents.processor.GatewayConfigurator;
+import com.redhat.service.smartevents.processor.GatewayValidator;
+
+abstract class BaseGatewayConstraintValidator<A extends Annotation, T> implements ConstraintValidator<A, T> {
```

Comment:
I validate a `Bridge`'s Error Handler `Action` so refactored some validation code to avoid duplication.